compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/BIRGen.java

```diff
@@ -494,12 +494,15 @@ public void visit(BLangClassDefinition classDefinition) {
 
         for (BAttachedFunction func : ((BObjectTypeSymbol) classDefinition.symbol).referencedFunctions) {
             BInvokableSymbol funcSymbol = func.symbol;
+            if (Symbols.isResource(funcSymbol) && Symbols.isFunctionDeclaration(funcSymbol)) {
```

Comment:
In what scenarios is the included function a resource declaration?

I tried the following but `((BObjectTypeSymbol) classDefinition.symbol).referencedFunctions` is empty at this point.

```ballerina
service class Foo {
    resource function get x() returns int {
        return 1;
    }
}

service class Bar {
    *Foo;

    resource function get x() returns byte {
        return 1;
    }
}
```

Reply:
Oh yes, this doesn't happen as we disallow resources decl.